# Assignment 12

# Chapter 7.1 - Compute an Average
## **Question 1. (screenshot: query and result)**
## Use table Sales.SalesOrderHeader to find the average freight per year

In [4]:
SELECT YEAR(OrderDate) As Year,
       AVG(Freight) AS AvgFreight
FROM AdventureWorks2022.Sales.SalesOrderHeader
GROUP BY Year(OrderDate);


(4 rows affected)

Total execution time: 00:00:00.032

Year AvgFreight 2013 89.7081 2014 47.8087 2011 224.3203 2012 252.4721

## **Question 2. (screenshot: query and result) - Order by year the Question 1 result.**

In [5]:
SELECT YEAR(OrderDate) As Year,
       AVG(Freight) AS AvgFreight
FROM AdventureWorks2022.Sales.SalesOrderHeader
GROUP BY Year(OrderDate)
ORDER BY Year;

(4 rows affected)

Total execution time: 00:00:00.021

Year AvgFreight 2011 224.3203 2012 252.4721 2013 89.7081 2014 47.8087

## **Question 3. (explanation) - Is the average freight generally increasing or decreasing?**
## The average freight has generally decreased over the years. Going from an average of 224.3 in 2011 to 47.8 in 2014

# 7.2 Count Rows in a Group
## **Question 4. (explanation) - What is the difference between COUNT and COUNT_BIG? Explain**

## The main distinction between COUNT\_BIG and COUNT is that the latter uses an integer to determine the total number of rows, which is typically adequate for most tables. COUNT\_BIG, on the other hand, handles really huge tables that can surpass the bounds of an integer by using a bigint, a data type that can store considerably larger values. For tables with many rows, COUNT\_BIG is the best option.

# 7.3 Sum Values in a Group
## **Question 5. (screenshot: query and result)**
## - Use the SUM function to add up a the TaxAmt of the Sales.SalesOrderHeader

In [1]:
SELECT SUM(TaxAmt) AS TotalTaxAmount
FROM AdventureWorks2022.Sales.SalesOrderHeader;

(1 row affected)

Total execution time: 00:00:48.043

TotalTaxAmount 10186974.4602

## **Question 6. (screenshot: query and result)**
## - Return only results for the year 2012, grouped and ordered by month.

In [2]:
SELECT YEAR(OrderDate) AS Year,
       MONTH(OrderDate) AS Month,
       SUM(TaxAmt) AS TotalTaxAmount
FROM AdventureWorks2022.Sales.SalesOrderHeader
WHERE 
    YEAR(OrderDate) = 2012
GROUP BY 
    YEAR(OrderDate),
    MONTH(OrderDate)
ORDER BY Month;

(12 rows affected)

Total execution time: 00:00:00.028

Year Month TotalTaxAmount 2012 1 371588.69 2012 2 132285.70 2012 3 274742.2648 2012 4 180817.2923 2012 5 288245.0511 2012 6 389556.4568 2012 7 321735.2997 2012 8 203286.8258 2012 9 325769.325 2012 10 239214.5416 2012 11 171010.3965 2012 12 264718.7376

# 7.4 Find High and Low Values in a Group
## **Question 7. (screenshot: query and result) - Find the Min TaxRate from the table Sales.SalesTaxRate.**

In [3]:
SELECT MIN(TaxRate) AS MinimumTaxRate
FROM AdventureWorks2022.Sales.SalesTaxRate;

(1 row affected)

Total execution time: 00:00:00.016

MinimumTaxRate 5.00

## **Question 8. (screenshot: query and result) - Find the Max TaxRate from the table Sales.SalesTaxRate**

In [4]:
SELECT MAX(TaxRate) AS MaximumTaxRate
FROM AdventureWorks2022.Sales.SalesTaxRate;

(1 row affected)

Total execution time: 00:00:00.004

MaximumTaxRate 19.60

# 7.6 Find Variance of Values in a Column
## **Question 9. (screenshot: query and result)**
## - Find the Variance of the TaxRate from the table Sales.SalesTaxRate

In [5]:
SELECT VAR(TaxRate) AS VarianceTaxRate
FROM AdventureWorks2022.Sales.SalesTaxRate;

(1 row affected)

Total execution time: 00:00:00.019

VarianceTaxRate 14.235548029556671

# 7.7 Find Std Dev of Values in a Column
## **Question 10. (screenshot: query and result)**
## - Find the StandardDeviation of the TaxRate from the table Sales.SalesTaxRate.


In [6]:
SELECT STDEV(TaxRate) AS StandardDeviationTaxRate
FROM AdventureWorks2022.Sales.SalesTaxRate;

(1 row affected)

Total execution time: 00:00:00.006

StandardDeviationTaxRate 3.7730025218062964